aiida/cmdline/commands/cmd_import.py

```diff
@@ -36,13 +37,108 @@ class ExtrasImportCode(Enum):
     ask = 'kca'
 
 
+def _try_import(first, **kwargs):
+    """
+    Utility function for `verdi import` to try-except to import archive
+
+    :param first: Boolean to determine what exception message to throw for IncompatibleArchiveVerisonError
+    """
+    from aiida.orm.importexport import import_data
+
+    # Checks
+    expected_keys = [
+        'file_to_import', 'archive', 'group', 'extras_mode_existing', 'extras_mode_new', 'comment_mode', 'migration',
+        'non_interactive'
+    ]
+    for key in expected_keys:
+        if key not in kwargs:
+            raise ValueError("{} needed for utility function '{}'".format(key, '_try_import'))
+
+    # Initialization
+    migrate_archive = False
+
+    try:
+        import_data(
+            kwargs['file_to_import'],
+            kwargs['group'],
+            extras_mode_existing=ExtrasImportCode[kwargs['extras_mode_existing']].value,
+            extras_mode_new=kwargs['extras_mode_new'],
+            comment_mode=kwargs['comment_mode'])
+    except exceptions.IncompatibleArchiveVersionError as exception:
+        if first:
+            # Migration has not yet been tried.
+            if kwargs['migration']:
+                # Confirm migration
+                echo.echo_warning(str(exception).splitlines()[0])
+                if kwargs['non_interactive']:
+                    migrate_archive = True
+                else:
+                    migrate_archive = click.confirm(
+                        "Do you want to try and migrate {} to the newest export file version?\n"
+                        "Note: This will not change your current file.".format(kwargs['archive']),
+                        default=True,
+                        abort=True)
+            else:
+                # Abort
+                echo.echo_critical(str(exception))
+        else:
+            # Migration has been performed, something is still wrong
+            crit_message = "{} has been migrated, but it still cannot be imported.\n{}".format(
+                kwargs['archive'], exception)
+            echo.echo_critical(crit_message)
+    except Exception:
+        echo.echo_error('an exception occurred while importing the archive {}'.format(kwargs['archive']))
+        echo.echo(traceback.format_exc())
+        if not kwargs['non_interactive']:
+            click.confirm('do you want to continue?', abort=True)
+    else:
+        echo.echo_success('imported archive {}'.format(kwargs['archive']))
+
+    return migrate_archive
+
+
+def _migrate_archive(ctx, temp_folder, **kwargs):
+    """Utility functiuon for `verdi import` to migrate archive"""
+    from aiida.cmdline.commands.cmd_export import migrate
+
+    # Checks
+    expected_keys = ['file_to_import', 'archive', 'non_interactive']
```

Comment:
why not simply make those keys explicit arguments?
that saves you from this check. you can still have "kwargs" to capture the rest

Reply:
For `_migrate_archive` it makes sense to make these few keys explicit. In `_try_import` there are so many, I thought it easier simply to do `**kwargs` for all of them. But sure, some of them, as in the keyword inputs to `import_data()` may simply be `**kwargs`. However, this is only three, making five explicit keyword inputs to `_try_import`, but maybe that is fine?

Reply:
Done